lua/neo-tree/sources/filesystem/lib/shared_clipboard.lua

```diff
@@ -0,0 +1,79 @@
+local fs_watch = require("neo-tree.sources.filesystem.lib.fs_watch")
+local manager = require("neo-tree.sources.manager")
+local events = require("neo-tree.events")
+local renderer = require("neo-tree.ui.renderer")
+local log = require("neo-tree.log")
+
+local M = {}
+
+local clipboard_state_dir_path = vim.fn.stdpath("state") .. "/neo-tree/"
+local clipboard_file_path = clipboard_state_dir_path .. "filesystem-clipboard.json"
+local clipboard_file_change_triggered_by_cur_neovim_instance = false
+
+M.save_clipboard = function(clipboard)
+  local file = io.open(clipboard_file_path, "w+")
+  -- We want to erase data in the file if clipboard is nil instead writing null
+  if not clipboard or not file then
+    return
+  end
+
+  local is_success, data = pcall(vim.json.encode, clipboard)
+  if not is_success then
+    log.error("Failed to save clipboard. JSON serialization error")
+    return
+  end
+  file:write(data)
```

Comment:
I actually meant to put the error handling n the `file:write` / `file:flush` code. That's where you are going to get frequent errors due to permissions, file locked, disk full, etc.